libs/python/interpreters_unix_test.go

```diff
@@ -0,0 +1,90 @@
+//go:build unix
+
+package python
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAtLeastOnePythonInstalled(t *testing.T) {
+	ctx := context.Background()
+	all, err := DetectInterpreters(ctx)
+	assert.NoError(t, err)
+	a := all.Latest()
+	t.Logf("latest is: %s", a)
+	assert.True(t, len(all) > 0)
+}
+
+func TestNoInterpretersFound(t *testing.T) {
+	t.Setenv("PATH", t.TempDir())
+
+	ctx := context.Background()
+	all, err := DetectInterpreters(ctx)
+	assert.Nil(t, all)
+	assert.Equal(t, ErrNoPythonInterpreters, err)
+}
+
+func TestFilteringInterpreters(t *testing.T) {
+	rogueBin := filepath.Join(t.TempDir(), "rogue-bin")
+	err := os.Mkdir(rogueBin, 0o777)
+	assert.NoError(t, err)
+	os.Chmod(rogueBin, 0o777)
+
+	raw, err := os.ReadFile("testdata/world-writeable/python8.4")
+	assert.NoError(t, err)
+
+	binary := filepath.Join(rogueBin, "python8.4")
+	err = os.WriteFile(binary, raw, 00777)
+	assert.NoError(t, err)
+
+	t.Setenv("PATH", "testdata/other-binaries-filtered"+string(os.PathListSeparator)+rogueBin)
```

Comment:
This should also test that `python8.4` is returned by `exec.LookPath`, otherwise we don't know for sure if the test setup is done correctly.